### PR TITLE
added 'addr' command which append the latest context and project tags…

### PR DESCRIPTION
**Pull request by mhalano**
added 'addr' command which append the latest context and project tags to a new item. I created using this add-on code:
https://github.com/ginatrapani/todo.txt-cli/wiki/Todo.sh-Add-on-Directory#addr--add-task-with-the-projects-and-contexts-of-the-last-added-task

### CODE REVIEW

**Comment by mhalano** [comment]
@karbassi Could you merge this if all changes are ok?

File: todo.sh [hunk 1]
```diff
@@ -175,6 +176,10 @@ actionsHelp()
 		      Adds SECOND THING I NEED TO DO to you todo.txt on its own line.
 		      Project and context notation optional.
 
+		    addr "THING I NEED TO DO +latestproject @latestcontext"
+		      Append the tags of the most recently added item
```
Comment (inkarkat) [inline todo.sh:180]: "Tags" is not part of the todo.txt vocabulary. These are always referred to as "projects and contexts".

File: todo.sh [hunk 2]
```diff
@@ -1113,6 +1118,50 @@ case $action in
...
+    else
+        [ -z "$2" ] && die "usage: $TODO_SH addr \"TODO ITEM\""
+        shift
+        input=$*
+    fi
+
+	# get the last line added and extract context
+	lastadded=$(sed -n '$ =' "$TODO_FILE")
+	# FIXME: Find a better way to add tasks (_addto function?)
+	"$TODO_FULL_SH" command add "$1"
```
Comment (inkarkat) [inline todo.sh:1134]: Yes, `_addto` should be used. Recursively invoking the script is not needed, and bad style.

File: todo.sh [hunk 3]
```diff
@@ -1113,6 +1118,50 @@ case $action in
...
+	newest=$(sed -n '$ =' "$TODO_FILE")
+
+	REGEX='[@|+][A-Za-z0-9]*'
```
Comment (inkarkat) [inline todo.sh:1137]: That is not the correct pattern to extract all projects and contexts. In fact, this now is configurable via `TODOTXT_SIGIL_VALID_PATTERN`. The `listWordsWithSigil()` function can extract these for you.

File: todo.sh [hunk 4]
```diff
@@ -1113,6 +1118,50 @@ case $action in
...
+
+	echo "Appending:" $(sed $lastadded"q;d" "$TODO_FILE" | grep -o "$REGEX")
+
+	# add contexts and projects
+	sed $lastadded"q;d" "$TODO_FILE" | grep -o "$REGEX" | while read -r app
+	do
+		echo $app
+	# FIXME: Find a better way to append content
```
Comment (inkarkat) [inline todo.sh:1145]: Again, don't do a recursive call. Either duplicate the `sed` call found in the `append` implementation, or (better) extract a new function and use that for both.

File: todo.sh [hunk 5]
```diff
@@ -1113,6 +1118,50 @@ case $action in
...
+	   	"$TODO_FULL_SH" command append "$newest" "$app"
+	done
+
+	# set priority
```
Comment (inkarkat) [inline todo.sh:1149]: Please remove commented-out code.

**Review by inkarkat — CHANGES_REQUESTED** [review]
There are still some loose ends here. Most importantly, you did not add any new tests yet that exercise the new command.

Also, while I see how this could indeed be useful, I doubt that the majority of todo.txt users would benefit greatly from this (because I haven't seen an existing add-on for this, and nobody has yet asked for this functionality) - but this is just my own opinion.

Things get added to the core todo.sh script because either these are hard or impossible to do with add-ons, or because an add-on has seen such widespread use that it makes sense to put this right into core.

So, the easy way out for you would be to turn this into an add-on (which should be trivial), and maybe share this with the community. The other alternative (provided that @karbassi is a bit more positive about the inclusion) would be to address the issues and add tests.

**Comment by karbassi** [comment]
Since @mhalano created an addon for this, let's keep it as an addon to limit updates to the core.

I'm open to discussion though, if anyone feels strongly about this.

**Pull request closed without merging**